### PR TITLE
Change instance config to use the correct Reserved IP parameter

### PR DESCRIPTION
**Pull request by haardikdharma10**
## Description

The CLI does not honour the `--publicip` flag during instance create. This PR fixes that.

### CODE REVIEW

File: cmd/instance/instance_create.go [hunk 1]
```diff
@@ -139,8 +139,9 @@ If you wish to use a custom format, the available fields are:
 			}
 		}
 
+		// Set public ipv4 if provided
 		if publicip != "" {
-			config.PublicIPRequired = publicip
+			config.ReservedIPv4 = publicip
```
Comment (alessandroargentieri) [inline cmd/instance/instance_create.go]: PublicIPRequired is either true o false and it is random and at no cost. It differs from the reserved IP which costs 3 dollars a month.

Comment (haardikdharma10) [reply cmd/instance/instance_create.go]: Yes we set that to "true" by default - https://github.com/civo/civogo/blob/master/instance.go#L190

File: cmd/instance/instance_create.go [hunk 2]
```diff
@@ -139,8 +139,9 @@ If you wish to use a custom format, the available fields are:
 			}
 		}
 
+		// Set public ipv4 if provided
```
Comment (alessandroargentieri) [inline cmd/instance/instance_create.go:142]:
```
    // Set reserved_ipv4 if provided
    if reservedIPv4 != "" {
         config.ReservedIPv4 = reservedIPv4
    }
```

Comment (haardikdharma10) [reply cmd/instance/instance_create.go:142]: Updated MR

File: cmd/instance/instance.go [hunk 3]
```diff
@@ -54,6 +54,7 @@ func init() {
 	instanceCreateCmd.Flags().StringVarP(&firewall, "firewall", "l", "", "the instance's firewall you can use the Name or the ID")
 	instanceCreateCmd.Flags().StringVarP(&tags, "tags", "g", "", "the instance's tags")
 	instanceCreateCmd.Flags().StringVarP(&privateIPv4, "private_ipv4", "", "", "Private IPv4 address")
+	instanceCreateCmd.Flags().StringVarP(&reservedIPv4, "reservedip", "", "", "Reserved IPv4 address")
```
Comment (alessandroargentieri) [inline cmd/instance/instance.go:57]: "reserved_ipv4" is the correct param to be passed to the api-go.
do we wanna map it as `--reserved-ipv4`?

**Review by alessandroargentieri — APPROVED** [review]
LGTM

**Pull request merged**